src/lib.rs

```diff
@@ -280,11 +298,9 @@ fn _core<'py>(_py: Python<'py>, m: &Bound<'py, PyModule>) -> PyResult<()> {
                     match uv_mode {
                         UVMode::Polarization => {
                             if (p.u * last_p.u + p.v * last_p.v) < 0.0.into() {
-                                p.u = -p.u;
-                                p.v = -p.v;
+                                p = -p;
                             }
-                            last_p.u = p.u;
-                            last_p.v = p.u;
```

Comment:
the defect was that `u` was stored instead of `v`. I refactored around it too so it's less obvious, but also less likely to come back.